libs/zedUpload/datastore.go

```diff
@@ -79,9 +79,16 @@ func httpClientSrcIP(localAddr net.IP, proxy *url.URL) *http.Client {
 	localTCPAddr := net.TCPAddr{IP: localAddr}
 	localUDPAddr := net.UDPAddr{IP: localAddr}
 	resolverDial := func(ctx context.Context, network, address string) (net.Conn, error) {
-		// XXX can DNS fallback to TCP? Would get a mismatched address if we do
-		d := net.Dialer{LocalAddr: &localUDPAddr}
-		return d.Dial(network, address)
+		switch network {
+		case "udp":
+			d := net.Dialer{LocalAddr: &localUDPAddr}
+			return d.Dial(network, address)
+		case "tcp":
+			d := net.Dialer{LocalAddr: &localTCPAddr}
+			return d.Dial(network, address)
```

Comment:
Should we use `"udp", "udp4", "udp6"` and `"tcp", "tcp4", "tcp6"`?

Reply:
I don't think that it is reachable.
See https://go.dev/src/net/dnsclient_unix.go, function `Resolver.exchange()`, lines 167-172:
```
var networks []string
if useTCP {
	networks = []string{"tcp"}
} else {
	networks = []string{"udp", "tcp"}
}
```

Reply:
But we can add it just in case...

Reply:
Done